ci-operator/step-registry/osd/grant/cluster-editor/osd-grant-cluster-editor-commands.sh

```diff
@@ -0,0 +1,181 @@
+  if [[ "${cluster_editor}" != "${username}" ]]; then
+    echo "Cluster editor name '${cluster_editor}' doesn't match the real username '${username}'"
+    return 1
+  fi
+}
+
+ocm_grant_cluster_editor_role() {
+  local cluster_id
+  local cluster_editor
+  local subscription_href
+  local request_json 
+  ocm_login
+  cluster_id="${1}"
+  cluster_editor="${2}"
+  subscription_href=$(ocm get "/api/clusters_mgmt/v1/clusters/${cluster_id}" | jq -r '.subscription.href')
+  request_json="{\"account_username\": \"${cluster_editor}\", \"role_id\": \"ClusterEditor\"}"
+  echo "Grant ClusterEditor role to ${cluster_editor}"
+  echo "${request_json}" | ocm post "${subscription_href}/role_bindings"
+}
+
+oc_share_cluster_editor_token() {
+  local kubeconfig_file
+  local cluster_editor
+  local cluster_editor_token
+  kubeconfig_file="${1}"
+  cluster_editor="${2}"
+  cluster_editor_token=$(read_profile_file "${CLUSTER_EDITOR_PREFIX}${cluster_editor}")
+  # make sure the namespace CLUSTER_SECRET_NS exists
+  echo "Create namespace ${CLUSTER_SECRET_NS}"
+  KUBECONFIG="${kubeconfig_file}" oc create ns "${CLUSTER_SECRET_NS}" --dry-run=client -o yaml | oc apply -f -
+  # create or update the OCM_TOKEN in CLUSTER_SECRET
+  echo "Create secret ${CLUSTER_SECRET} with key OCM_TOKEN"
+  KUBECONFIG="${kubeconfig_file}" oc create secret generic "${CLUSTER_SECRET}" --namespace="${CLUSTER_SECRET_NS}" --from-literal=OCM_TOKEN="${cluster_editor_token}" --dry-run=client -o yaml | oc apply -f -
+}
+
+main() {
+  local cluster_id
+  local cluster_editor
+  local cluster_editors
+  cluster_id=$(get_cluster_id "CLUSTER_ID" "${SHARED_DIR}/cluster-id")
+  if [[ ! -n "${cluster_id}" ]]; then
+    error "Cluster id is not defined!"
+  fi
+  echo "Cluster id: ${cluster_id}"
+
+  local kubeconfig="${SHARED_DIR}/kubeconfig"
+  if [[ ! -n "${kubeconfig}" ]]; then
+    error "Cannot find kubeconfig!"
+  fi
+  echo "Kubeconfig: ${kubeconfig}"
+
+  echo "Cluster editor prefix: ${CLUSTER_EDITOR_PREFIX}"
+  cluster_editors=$(get_cluster_editors "${CLUSTER_PROFILE_DIR}" "${CLUSTER_EDITOR_PREFIX}")
+  echo "Cluster editors: ${cluster_editors:-no cluster editors found}"
+
+  echo "Cluster editor strategy: ${CLUSTER_EDITOR_STRATEGY}"
+  cluster_editor=$(select_cluster_editor "${cluster_editors}" "${CLUSTER_EDITOR_STRATEGY}")
+  echo "Cluster editor: ${cluster_editor:-no cluster editor was selected}"
+
+  local cluster_editor_check
+  if [[ -n "${cluster_editor}" ]]; then
+    cluster_editor_check=$(ocm_check_cluster_editor "${cluster_editor}")
+     if [[ -n "${cluster_editor_check}" ]]; then
+      error "${cluster_editor_check}"
+    fi
+    # Grant ClusterEditor role on OSD via ocm
+    ocm_grant_cluster_editor_role "${cluster_id}" "${cluster_editor}"
+    # Share the appropriate secret on openshift via oc
+    oc_share_cluster_editor_token "${kubeconfig}" "${cluster_editor}"
+  fi
+}
+
+# Sourceable from *test.sh files
+if [[ ${0} != *"test.sh" ]]; then
```

Comment:
@apodhrad What is the functionality of test.sh ? 
Where is it inside the PR?

Reply:
Further, should we be providing an option to customers to skip this test all together ?

Reply:
Hi @samvarankashyap , I have an external test script (with suffix `*test.sh`) which test particular functions (unit tests). Unfortunately, PR checks failed when I added the test script there :(

I planned to investigate the issue but until that I wanted someone to review this PR.

Reply:
IMHO, the part
```
if [[ ${0} != *"test.sh" ]];
```
can be kept as it skips the auto execution of the main functionality if and only if `osd-grant-cluster-editor-commands.sh` is sources from a file with suffix `*test.sh`.